lib/scope.js

```diff
@@ -668,19 +681,66 @@ module.exports = {
         // TODO: Text inputs and textareas must ignore the value, etc. How?
         values.push( node.attribs.value );
       }
+      // TODO: add default of empty string for values?
+
+      field_data.var_names = var_names;
+      field_data.values = values || [];
+
+      // We need to try the full names of each field. If there are proxy vars (x, i, etc)
+      // this will let us find which one matches the sought_var and we can use it to
+      // create a map from proxies to intrinsic names. Ex: {x: "foo"}.
+      // We will then need to use the data gathered here to convert each field to
+      // have its intrinsic name.
+      field_data.name_patterns = [];
+      for ( let name of var_names ) {
+        // Get the data that will help us construct the intrinsic name later and
+        // look for the sought_var match now.
+        let pattern = await scope.getSoughtVarPatterns( scope, { field_name: name, });
+        field_data.name_patterns.push( pattern );
+        // Try to pull out the intrinsic name parts
+        let maybe_map = await scope.getProxiesToNamesMap( scope, {
+          sought_var: page_data.sought_var, pattern_data: pattern,
+        });
+        // If that works, it's the right map for the proxy values
+        if ( maybe_map ) { proxies_to_names_map = maybe_map; }
```

Comment:
I don't understand the advantage of using `maybe_map` here instead of just assigning directly to `proxies_to_names_map`

Reply:
Good question! #196! Though having trouble of thinking of a succinct way to document it.